File: Backend/api-order/src/main/java/com/ms/order/model/Order.java
package com.ms.order.model;

import com.ms.order.dto.OrderDTO;
import com.ms.order.dto.OrderItemDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import org.springframework.beans.BeanUtils;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.List;

@Data
@AllArgsConstructor
@Builder
@Document(collection = "orders")
public class Order implements Serializable {
    @Id
    private String id;
    private String client;
    private String userId;
    private String orderStatus;
    private List<OrderItemDTO> orderItems;
    private String paymentMethod;
    private String paymentInformation;
    private String orderNotes;
    private String created;
    private String updated;
    private String status;
    public Order(OrderDTO orderDTO){
        BeanUtils.copyProperties(orderDTO, this);
    }

    public Order() {
        super();
    }

}

File: Frontend/ogani-site/src/app/pages/order/model/order.ts
export class Order {
  id?: string;
  client?: string;
  userId!: string;
  orderItems: OrderItem[];
  paymentInformation!: string;
  orderStatus!: string;
  paymentMethod!: string;
  orderNotes!: string;
  created!: string;
  updated!: string;

  constructor(data: Partial<Order> = {}) {
    Object.assign(this, data);
    if (!this.orderItems) {
      this.orderItems = [];
    }
  }
}

export interface OrderItem {
  idProduct: string;
  name: string;
  description: string;
  quantity: number;
  price: number;
  discount: number;
}
